x-pack/plugins/observability_solution/apm/public/components/alerting/ui_components/alert_details_app_section/latency_alerts_history_chart.tsx

```diff
@@ -65,7 +67,11 @@ export function LatencyAlertsHistoryChart({
     end,
     kuery: '',
     numBuckets: 100,
-    type: ApmDocumentType.ServiceTransactionMetric,
+    // ServiceTransactionMetric does not have transactionName as a dimension, but it is faster than TransactionMetric
+    // We use TransactionMetric only when there is a transactionName
+    type: transactionName
+      ? ApmDocumentType.TransactionMetric
+      : ApmDocumentType.ServiceTransactionMetric,
```

Comment:
Based on this [comment](https://github.com/elastic/kibana/pull/180558/files#r1561079278)